.github/workflows/merge-blocking-pr.yml

```diff
@@ -0,0 +1,54 @@
+name: Merged Blocking Pull Request Automation
+
+on:
+  pull_request:
+    types:
+      - closed
+
+jobs:
+  update-blocked-status:
+    name: Update Blocked Status
+    runs-on: ubuntu-latest
+
+    # a pr that was a `blocking:<id>` label was merged.
+    # find the open pr's it was blocked by and trigger a refresh of their state
+    if: github.event.pull_request.merged == true && contains( join( github.event.pull_request.labels.*.name, ',' ), 'blocking' )
+
+    steps:
+      - name: Generate token
+        id: generate-token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ vars.PULL_REQUEST_APP_ID }}
+          private-key: ${{ secrets.PULL_REQUEST_APP_PRIVATE_KEY }}
+
+      - name: Gather Dependent PRs
+        id: gather_deps
+        env:
+          GH_TOKEN: ${{ steps.generate-token.outputs.token }}
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+        run: |
+          blocked_prs=$(
+          gh -R ${{ github.repository }} pr list --label 'blocked' --json 'number,body' \
+            | jq -c --argjson pr "${{ github.event.pull_request.number }}" '
+                reduce ( .[] | select(
+                  .body |  
+                    scan("(?:blocked (?:by|on)|stacked on):? #([0-9]+)") |
+                    map(tonumber) |
+                    any(.[]; . == $pr)
+                )) as $i ([]; . + [$i])
+              '
+          )
+          echo "deps=$blocked_prs" >> "$GITHUB_OUTPUT"
+          echo "numdeps=$(jq -r '. | length' <<< "$blocked_prs")" >> "$GITHUB_OUTPUT"
```

Comment:
Maybe we could re-use the status we set instead? That should be a lot more consistent in consumption than scanning (though it isn't that big of a deal since this is obv the same logic we use to apply that status in the first place)

Reply:
the trick is that this only triggers reevaluation for the Prs that *directly* depended on the merged Pr. if we only went off the label we could trigger dozens of unneeded workflows.